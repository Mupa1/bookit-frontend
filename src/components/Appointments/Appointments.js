import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';

import Appointment from '../Appointment/Appointment';
import Sidebar from '../Sidebar/Sidebar';
import { getAppointments } from '../../Redux/actions/index';

const Appointments = ({
  appointment, getAppointments, user,
}) => {
  useEffect(async () => {
    await fetch('http://localhost:3001/api/v1/appointments', {
      mode: 'cors',
      headers: {
        'Content-Type': 'application/json',
        'AUTH-TOKEN': localStorage.getItem('token'),
      },
    }).then(res => res.json())
      .then(data => getAppointments(data.data.appointment))
      .catch(error => (error));
  }, []);

  if (Object.keys(user).length === 0) { return <Redirect to="/" />; }

  return (
    <>
      <Sidebar />
      {appointment ? (
        <section className="content">
          <h2 className="font-weight-bold text-center pb-5">APPOINTMENTS</h2>
          <div className="table-responsive">
            <table className="container table table-striped">
              <thead>
                <tr>
                  <th>Doctor&apos;s Name</th>
                  <th>Username</th>
                  <th>Date</th>
                  <th>City</th>
                </tr>
              </thead>
              <tbody>
                {appointment.map(appoint => (
                  <Appointment
                    key={appoint.id}
                    appointment={appoint}
                  />
                ))}
              </tbody>
            </table>
          </div>
        </section>
      ) : (
        <section className="content text-center">
          <h4 className="py-5">No appointments yet!</h4>
        </section>
      )}
    </>
  );
};

Appointments.propTypes = {
  appointment: PropTypes.arrayOf(PropTypes.object).isRequired,
  getAppointments: PropTypes.func.isRequired,
  user: PropTypes.instanceOf(Object).isRequired,
};

const mapStateToProps = state => ({
  appointment: state.appointment,
  user: state.user,
});

const mapDispatchToProps = dispatch => ({
  getAppointments: appointment => {
    dispatch(getAppointments(appointment));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Appointments);
