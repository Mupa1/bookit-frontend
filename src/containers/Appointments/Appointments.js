import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Appointment from '../../components/Appointment/Appointment';
import Sidebar from '../Sidebar/Sidebar';
import { fetchAppointments } from '../../api';

const Appointments = () => {
  const dispatch = useDispatch();
  const appointments = useSelector(state => state.appointments.appointments);

  useEffect(() => {
    dispatch(fetchAppointments());
  }, []);

  return (
    <>
      <Sidebar />
      {appointments.length ? (
        <section className="content">
          <h2 className="font-weight-bold text-center pb-5">APPOINTMENTS</h2>
          <div className="table-responsive">
            <table className="container table table-striped">
              <thead>
                <tr>
                  <th>Doctor&apos;s Name</th>
                  <th>Username</th>
                  <th>Date</th>
                  <th>City</th>
                </tr>
              </thead>
              <tbody>
                {appointments.map(appointment => (
                  <Appointment
                    key={appointment.id}
                    appointment={appointment}
                  />
                ))}
              </tbody>
            </table>
          </div>
        </section>
      ) : (
        <section className="content text-center">
          <h4 className="py-5">No appointments yet!</h4>
        </section>
      )}
    </>
  );
};

export default Appointments;
