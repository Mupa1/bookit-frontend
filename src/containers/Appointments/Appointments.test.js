import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Appointments from './Appointments';
import store from '../../Redux/store';

describe('Appointments', () => {
  it('renders Appointments component', () => {
    render(
      <Provider store={store}>
        <Router>
          <Appointments />
        </Router>
      </Provider>,
    );
  });

  it('matches Appointments snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Appointments />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
