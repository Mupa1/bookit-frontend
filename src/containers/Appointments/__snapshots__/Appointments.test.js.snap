// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments matches Appointments snapshot 1`] = `null`;
